packages/server/src/mail.ts

```diff
@@ -0,0 +1,134 @@
+import fs from 'fs';
+import handlebars from 'handlebars';
+import nodemailer from 'nodemailer';
+import path from 'path';
+
+import { config } from '~/config';
+import { Currency, Customer, Invoice } from '~/entities';
+import { formatDate } from '~/lib/dayjs';
+import { log } from '~/log';
+
+let transporter: nodemailer.Transporter;
+
+export function init(): void {
+  const mailConfig = config.mail;
+
+  if (!mailConfig.host || !mailConfig.port || !mailConfig.username || !mailConfig.password) {
+    log.info('No mail config found, skipping mail initialization');
+    return;
+  }
+
+  handlebars.registerHelper('formatDate', (date: Date, format: string) => formatDate(date, format));
+  handlebars.registerHelper('amountToPrice', (amount: number, currency: Currency) =>
+    Invoice.amountToPrice(amount, currency),
+  );
+
+  transporter = nodemailer.createTransport({
+    host: mailConfig.host,
+    port: mailConfig.port,
+    secure: mailConfig.secure,
+    requireTLS: mailConfig.requireTLS,
+    auth: {
+      user: mailConfig.username,
+      pass: mailConfig.password,
+    },
+  });
+}
+
+type Languages = 'en' | 'de';
+type Template = {
+  subject: Record<Languages, ReturnType<typeof handlebars.compile>>;
+  text: Record<Languages, ReturnType<typeof handlebars.compile>>;
+};
+
+const templates: Record<string, Template> = {
+  newInvoice: {
+    subject: {
+      en: handlebars.compile(`{{ project.name }} - New invoice {{ invoice.number }}`),
+      de: handlebars.compile(`{{ project.name }} - Neue Rechnung {{ invoice.number }}`),
+    },
+    text: {
+      en: handlebars.compile(`
```

Comment:
@anbraten For typesafety we should do
```suggestion
      en: handlebars.compile<{ customer: Customer; invoice: Invoice }>(`
```

here and also at the de template

Reply:
Wasn't aware that's possible. Nice one.

Reply:
@mariusheine Tried that, but it then complains that the type of the Record is not correct. So we either need to remove the Record type or ignore the generic.

Reply:
@anbraten I played a bit around and came up with a bit complex but quite well working type. (or a few more types :rofl:) What do you think about it?

Reply:
@anbraten 33ac5d7ee2745f76e88f6facbc7b17d08588e096

Reply:
@anbraten Ok I had to fix some checks but now it is totally fine, but they are two commits now ;)

Reply:
@mariusheine Quite some code for a "simple" type, but fine for me.